devicetypes/smartthings/fibaro-smoke-sensor.src/fibaro-smoke-sensor.groovy

```diff
@@ -119,9 +119,28 @@ metadata {
 
 def updated() {
 	log.debug "Updated with settings: ${settings}"
+	if(!state.legacySettingsUpdated) updateLegacySettings()
 	setConfigured("false") //wait until the next time device wakeup to send configure command
 }
 
+def updateLegacySettings() {
+
+	def legacyNotificationOptionMap = {
+		"disabled" : "None",
+		"casing opened" : "Casing opened",
+		"exceeding temperature threshold" : "Exceeding temperature threshold",
+		"lack of Z-Wave range" : "Lack of Z-Wave range",
+		"all notifications" : "All"
+	}
+
+	temperatureReportInterval == "reports inactive" ? "Reports inactive" : temperatureReportInterval
```

Comment:
This still needs to be updated?

Reply:
this is how the variable is being referenced in the configure method, so I assume it works, but will verify.

Reply:
My point is that it's checking a condition but not doing an assignment